pkg/kubelet/kubelet.go

```diff
@@ -869,47 +870,53 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *api.Pod, container *api.Contain
 		return result, err
 	}
 
-	for _, value := range container.Env {
+	// Determine the final values of variables in steps:
+	//
+	// 1.  Determine the final value of each variable which is not sourced by expanding `$(var)`
+	//     references to other variables in the .Value field; the sources of variables are the
+	//     declared variables of the container and the service environment variables
+	// 2.  If a source is defined for an environment variable, resolve the source
```

Comment:
These first two steps are confusing:
In 1, you say determine the final value of each variable which is not sourced by expanding $() references, but it looks like the expanding is being done.
Also, the comment makes it sound like 1 and 2 are separate ordered steps, but they're really just two alternatives to the same step